vita3k/gui/src/profiles_manager_dialog.cpp

```diff
@@ -69,106 +80,83 @@ void draw_profiles_manager_dialog(GuiState &gui, HostState &host) {
     } else
         ImGui::PopStyleColor();
 
-    // User list
-    ImGui::PushItemWidth(300);
-    if (ImGui::ListBoxHeader("##online_id_list", (int)host.cfg.online_id.size(), 4)) {
-        for (auto i = 0; i < host.cfg.online_id.size(); i++) {
-            ImGui::TextColored(GUI_COLOR_TEXT, "User ID: %02d, Online ID: %s", i, host.cfg.online_id[i].c_str());
-        }
+    // Users list
+    ImGui::TextColored(GUI_COLOR_TEXT, "Select Your User.");
+    if (ImGui::ListBoxHeader("##users_list", static_cast<int>(host.cfg.users.size()), 6)) {
+        for (const auto &u : host.cfg.users)
+            if (ImGui::Selectable(u.second.c_str(), u.first == host.cfg.current_user_id))
+                host.cfg.current_user_id = u.first;
         ImGui::ListBoxFooter();
     }
-    ImGui::PopItemWidth();
+    ImGui::SameLine();
+    ImGui::TextColored(GUI_COLOR_TEXT, "Current user ID: %02d", host.cfg.current_user_id);
     ImGui::Spacing();
-    if (host.cfg.online_id.size() > 1) {
-        ImGui::PushItemWidth(210);
-        ImGui::TextColored(GUI_COLOR_TEXT, "Select Your User.");
-        ImGui::Combo(
-            "##select_user", &host.cfg.user_id,
-            [](void *vec, int idx, const char **list_online_id) {
-                auto &vector = *static_cast<std::vector<std::string> *>(vec);
-                if (idx < 0 || idx >= static_cast<int>(vector.size())) {
-                    return false;
-                }
-                *list_online_id = vector.at(idx).c_str();
-                return true;
-            },
-            static_cast<void *>(&host.cfg.online_id), static_cast<int>(host.cfg.online_id.size()));
-        ImGui::SameLine();
-        ImGui::TextColored(GUI_COLOR_TEXT, "User ID: %02d", host.cfg.user_id);
-        ImGui::PopItemWidth();
 
-        // Delete User
-        ImGui::Spacing();
-        if (ImGui::Button("Delete Selected User"))
+    // Delete User
+    if (host.cfg.users.size() > 1) {
+        std::string delete_button = "Delete Main User";
+        if (host.cfg.current_user_id > host.cfg.users.begin()->first)
```

Comment:
what is this supposed to do?

Reply:
Is simple change name button if is your first user ornif is other user